### PR TITLE
drone: 0.8.6 -> 1.3.1

### PR DESCRIPTION
**Pull request by elohmeier**
###### Motivation for this change
update to latest stable version

###### Things done
switched to buildGoModule, targeted latest released stable version

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @avnik @vdemeester

### CODE REVIEW

File: pkgs/development/tools/continuous-integration/drone/default.nix [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
-
-  nativeBuildInputs = [ go-bindata go-bindata-assetfs ];
+  modSha256 = "09mg1lrq93jpl98yp51mpqziarl5dgwm3ifi7f16lchvyy192k3f";
 
   src = fetchFromGitHub {
     owner = "drone";
     repo = "drone";
-    rev = revision;
-    sha256 = "0miq2012nivvr1ysi3aa2xrr5ak3mf0l3drybyc83iycy0kp4bda";
+    rev = "e1ed2623b15e1ab77a468f07cf28307423c0fc88";
```
Comment (rasendubi) [inline pkgs/development/tools/continuous-integration/drone/default.nix]: Use `v1.3.1` tag instead:
```suggestion
    rev = "v${version}";
```

Comment (avnik) [reply pkgs/development/tools/continuous-integration/drone/default.nix]: I agree about using tagged revision, if it possible here.

Also if one of you want to overtake maintainership of drone, you are welcomed. I not use it actively at the moment.

**Comment by risicle** [comment]
On macos 10.13:

```
hash mismatch in fixed-output derivation '/nix/store/pv8dplcwn6yg7qzi1i0dx5qscrn12bmj-drone.io-1.3.1-go-modules':
  wanted: sha256:09mg1lrq93jpl98yp51mpqziarl5dgwm3ifi7f16lchvyy192k3f
  got:    sha256:128nnn8axnr30y3r46d11g8qznbsg82z2qw9qbxkq3bja6nz8fn2
cannot build derivation '/nix/store/hyp6420dgbcw3abh4n5f04jbdlwqn64a-drone.io-1.3.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r43fl2f7nx2876p2zcg2pm37kklpjzlq-env.drv': 1 dependencies couldn't be built
```

File: pkgs/development/tools/continuous-integration/drone/default.nix (hunk 1 above)
Comment (elohmeier) [reply pkgs/development/tools/continuous-integration/drone/default.nix]: Thanks for your feedback, I've switched to a tagged revision and updated to the maintainers list.

**Comment by elohmeier** [comment]
@risicle, thanks for your feedback. I've fixed the checksum.

**Review by vdemeester — APPROVED** [review]
LGTM

**Comment by rasendubi** [comment]
@GrahamcOfBorg build drone

**Comment by matthiasbeyer** [comment]
Can someone merge this?

**Review by avnik — APPROVED** [review]
LGTM.  CI said it's buildable (and I will check myself today later)

**Comment by c0bw3b** [comment]
(retrying darwin build)
@GrahamcOfBorg build drone

**Comment by c0bw3b** [comment]
Darwin :heavy_check_mark: so merging.

@elohmeier there's already newer versions upstream. `drone-cli` could also use a bump.
Feel free to submit another PR if you want to update those in nixpkgs. :)

**Comment by matthiasbeyer** [comment]
Also feel free to submit a service definition and close #70519

**Pull request merged**